### PR TITLE
[ModelRunner] Make leak sanitizer happy

### PR DESCRIPTION
**Pull request by qcolombet**
When running the model-runner, we don't explicitly delete the loader
that we dynamically allocate. Although the memory will be reclaimed
at the end of the process, this made leak sanitizer to complain.

NFC.

### CODE REVIEW

File: include/glow/Base/Tensor.h [hunk 1]
```diff
@@ -243,6 +243,9 @@ class Tensor final {
 
     if (size()) {
       size_t count = size() * type_.getElementSize();
+      // We are allocating memory specifically for this tensor,
+      // thus, it owns it.
+      isUnowned_ = false;
```
Comment (rdzhabarov) [inline include/glow/Base/Tensor.h]: can you reproduce this in a test so that we have sanitizer to complain about this issue?

Comment (qcolombet) [reply include/glow/Base/Tensor.h]: Added a test case in the updated PR.

File: tests/unittests/tensorsTest.cpp [hunk 2]
```diff
@@ -399,6 +399,34 @@ TEST(Tensor, nonOwnedTensor) {
...
 
+/// Check that we properly take ownership of
+/// the underlying memory when we reset the tensor
+/// shape. This test used to fail leak sanitizer.
+TEST(Tensor, nonOwnedTensorFollowedByReset) {
+  uint8_t raw_data[] = {0, 0, 0, 0};
+  Type F32Ty(ElemKind::FloatTy, {1});
+
+  // Create an unowned tensor.
+  Tensor T1(raw_data, &F32Ty);
```
Comment (jfix71) [inline tests/unittests/tensorsTest.cpp]: nit: Since this is `FloatTy`, why aren't you just creating the raw_data as `float raw_data[] = {0.};`?

Comment (qcolombet) [reply tests/unittests/tensorsTest.cpp]: Good point! I guess I was too attached as representing the memory byte by byte :P

**Pull request merged**